src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWebCSharp.1/Program.cs

```diff
@@ -95,13 +95,14 @@
 #endif
 {
     app.UseExceptionHandler("/Error", createScopeForErrors: true);
-    app.UseStatusCodePagesWithReExecute("/Error", createScopeForStatusCodePages: true);
 #if (HasHttpsProfile)
     // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
     app.UseHsts();
 #endif
 }
 
+app.UseStatusCodePagesWithReExecute("/Error", createScopeForStatusCodePages: true);
```

Comment:
It seems you have problems with opening links. Change this to

app.UseStatusCodePagesWithReExecute("/not-found", createScopeForStatusCodePages: true);

Also, edit the file 

src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWebCSharp.1/Program.Main.cs

by adding the same code in line 110.